pkg/security/utils/interner.go

```diff
@@ -0,0 +1,37 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package utils
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/util/intern"
+)
+
+// StringInterner is a best-effort LRU-based string deduplicator
+type StringInterner struct {
+	interner *intern.StringInterner
+}
+
+// NewStringInterner returns a new LRUStringInterner, with the cache size provided
+// if the cache size is negative this function will panic
+func NewStringInterner() *StringInterner {
+	return &StringInterner{
+		interner: intern.NewStringInterner(),
+	}
+}
+
+// DeduplicateBytes returns a possibly de-duplicated string from raw bytes
+func (si *StringInterner) DeduplicateBytes(value []byte) string {
+	return si.interner.Get(value).Get()
```

Comment:
I'm not sure this is going to do much, since you don't hold onto the pointers. They will get garbage collected. The idea is to use the pointer value in place of the `string` until you actually need the string value. The pointer values are also smaller than a string header (8 bytes vs 16 bytes).

Reply:
In this case this will never work, and I will go back to the LRU implementation, there is simply no way we change our whole code base to use those it's way too much churn